metric/meter.go

```diff
@@ -116,7 +118,15 @@ type Meter interface {
 // the same attributes as another Callback will report.
 //
 // The function needs to be concurrent safe.
-type Callback func(context.Context) error
+type Callback func(context.Context, MultiObserver) error
+
+// MultiObserver records measurements for multiple instruments in a Callback.
+type MultiObserver interface {
+	// Float64 records the float64 value with attributes for obsrv.
+	Float64(obsrv instrument.Float64Observer, value float64, attributes ...attribute.KeyValue)
```

Comment:
How about ObserveFloat64?  It feels more readable to have 

```golang
o.ObserveFloat64(mygauge, 1.0)
```
vs 
```golang
o.Float64(mygauge, 1.0)
```

Reply:
Sure :smile:

I tried following your logic from the other comment on shorter is better, but might have over-done it.

Reply:
I know I was contradicting myself... Shortest understandable name is what I'm going for :).